tools/refguide_check.py

```diff
@@ -150,43 +151,155 @@ def report(all_dict, funcnames, deprecated, module_name):
                 print(name)
 
 
-def check_docstrings(module):
-    """Check the code in the docstrings of the module's public symbols.
+def check_docstrings(module, verbose):
+    """Check code in docstrings of the module's public symbols.
     """
+    # these names are known to fail doctesting and we like to keep it that way
+    # e.g. sometimes pseudocode is acceptable etc
+    skiplist = set(['quad', 'UnivariateSpline', 'levy_stable'])
 
-    class DTRunner(doctest.DocTestRunner):
-        def report_failure(self, out, test, example, got):
-            # do not complain if output does not match
-            pass
-
-    # namespace to run examples in
+    # the namespace to run examples in
     ns = {'np': np,
           'assert_allclose': np.testing.assert_allclose,
-          'assert_equal': np.testing.assert_equal}
+          'assert_equal': np.testing.assert_equal,
+          # recognize numpy repr's
+          'array': np.array,
+          'int64': np.int64,
+          'uint64': np.uint64,
+          'int8': np.int8,
+          'int32': np.int32,
+          'float64': np.float64,
+          'dtype': np.dtype,
+          'nan': np.nan, 'NaN': np.nan,
+          'inf': np.inf, 'Inf': np.inf,}
+
+    # if MPL is available, use display-less backend
+    try:
+        import matplotlib
+        matplotlib.use('Agg')
+        import matplotlib.pyplot as plt
+        have_MPL = True
+    except ImportError:
+        have_MPL = False
+
+
+    class DTRunner(doctest.DocTestRunner):
+        stopwords = {'plt.', '.hist', '.show', '.ylim', '.subplot(',
+                     'set_title', 'imshow', 'plt.show', 'ax.axis', 'plt.plot(',
+                     '.title', '.ylabel', '.xlabel', 'set_ylim', 'set_xlim'}
+        rndm_markers = {'# random', '# Random', '#random', '#Random'}
+
+        def report_failure(self, out, test, example, got):
+            if (any(word in example.source for word in self.stopwords) or
```

Comment:
Might be more robust to do this by checking if `out` is some matplotlib object and contains
something like `<class 'matplotlib.`?